cloud/src/autogluon/cloud/predictor/cloud_predictor.py

```diff
@@ -0,0 +1,876 @@
+        return os.path.abspath(path)
+
+    def _setup_cloud_output_path(self, path):
+        if path is None:
+            return f's3://ag-cloud-predictor/{sagemaker.utils.sagemaker_timestamp()}'
+        if path.endswith('/'):
+            path = path[:-1]
+        if is_s3_url(path):
+            return path
+        return 's3://' + path
+
+    def _retrieve_latest_framework_version(self, framework_type='training'):
+        return retrieve_latest_framework_version(framework_type)
+
+    def _parse_framework_version(self, framework_version, framework_type):
+        if framework_version == 'latest':
+            framework_version = self._retrieve_latest_framework_version(framework_type)
+        else:
+            valid_options = retrieve_available_framework_versions(framework_type)
+            assert framework_version in valid_options, f'{framework_version} is not a valid option. Options are: {valid_options}'
+        return framework_version
+
+    def _construct_config(self, predictor_init_args, predictor_fit_args, leaderboard):
+        assert self.predictor_type is not None
+        config = dict(
+            predictor_type=self.predictor_type,
+            predictor_init_args=predictor_init_args,
+            predictor_fit_args=predictor_fit_args,
+            leaderboard=leaderboard,
+        )
+        path = os.path.join(self.local_output_path, 'utils', 'config.yaml')
+        with open(path, 'w') as f:
+            yaml.dump(config, f)
+        return path
+
+    def _setup_bucket(self, bucket):
+        s3 = boto3.resource('s3')
+        if not s3.Bucket(bucket) in s3.buckets.all():
+            s3.create_bucket(
+                Bucket=bucket,
+                CreateBucketConfiguration={
+                    'LocationConstraint': self.sagemaker_session.boto_region_name,
+                }
+            )
+
+    # FIXME: Remember to change output_type back to parquet when parquet is fixed in the gpu container
+    def _prepare_data(self, data, filename, output_type='csv'):
+        assert output_type in ['parquet', 'csv'], f'output type:{output_type} is not supported'
+        path = os.path.join(self.local_output_path, 'utils')
+        converter = FormatConverterFactory.get_converter(output_type)
+        return converter.convert(data, path, filename)
+
+    def _upload_fit_artifact(
+        self,
+        train_data,
+        tune_data,
+        config
+    ):
+        cloud_bucket, cloud_key_prefix = s3_path_to_bucket_prefix(self.cloud_output_path)
+        util_key_prefix = cloud_key_prefix + '/utils'
+        train_input = train_data
+        if isinstance(train_data, pd.DataFrame) or not is_s3_url(train_data):
+            train_data = self._prepare_data(train_data, 'train')
+            logger.log(20, 'Uploading train data...')
+            train_input = self.sagemaker_session.upload_data(
+                path=train_data,
+                bucket=cloud_bucket,
+                key_prefix=util_key_prefix
+            )
```

Comment:
For the large inputs (>4GB) we need to do [multipart uploads](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/s3.html#multipart-transfers). 
The additional parameters can be passed via `ExtraArgs` - [code](https://github.com/aws/sagemaker-python-sdk/blob/11ce41808690ef71144d3ca7deb3c2f4db318852/src/sagemaker/session.py#L233).
On other hand if we decide not to support very large inputs, we should throw an error if the file is to big.

Reply:
After reading some docs, I think this multipart uploads is enabled by default: [here](https://boto3.amazonaws.com/v1/documentation/api/latest/_modules/boto3/s3/transfer.html#S3Transfer.upload_file). The S3Transfer object initialize a default config, which has a default multipart_threshold==8MB